portal/academy/services.py

```diff
@@ -0,0 +1,29 @@
+import logging
+
+from portal.academy.models import Specialization, Grade, Unit
+from portal.users.models import User
+
+
+logger = logging.getLogger(__name__)
+
+
+def check_complete_specialization(user: User, spec: Specialization):
+    """
+    Check student completed a specialization by verifying
+    they passed on all units
+    """
+    logger.info(f"Checking {user.name} completion of specialization: {spec.name}")
+
+    spec_units = [unit for unit in Unit.objects.filter(specialization=spec)]
+    grades = Grade.objects.filter(user=user, unit__in=spec_units)
+    units = [g.unit for g in grades]
+    scores = [g.score for g in grades]
+
+    logger.info(f"Student {user.name} has following scores in {spec.name}: {scores} ({units})")
+
+    if len(scores) == len(spec_units) and all(score is not None and score >= 16 for score in scores):
```

Comment:
Doesn't this assume that there's only one grande per unit?
The student will have one grade instance per submission.

Reply:
Ah, thanks. So I should take the max I guess? Do we store the best submission inside deadline anywhere or do we just do that heuristic on top of the grades? Can you point me to the correct place to look at that logic? :)

Reply:
We currently only display the last grade, so it isn't in the code yet.
I suppose it would make sense to be part of #79, which *someone* hasn't started yet.

Reply:
Ok, thanks :) 

1. I changed the current logic to assess the max_score
2. I added a TODO so that when #79 is closed and we have more info on the grade we use only the top grade before submission here
3. Added tests representing situations where more than one grade exists for some units to ensure the logic can't be reversed back to the old one 

Let me know if it's ok now @hcastilho